Fix JSON serialization issues

stxrtt opened the pull request:
Replaced manual string concatenation with JsonConvert.SerializeObject for proper JSON formatting. Prevents potential formatting errors and simplifies the payload creation.

stxrtt committed:
Fix JSON serialization issues

Replaced manual string concatenation with JsonConvert.SerializeObject for proper JSON formatting.
Prevents potential formatting errors and simplifies the payload creation.

UnityGameStudio commented:
Be aware, your DeepSeek token was saved with the Unity Package! Indeed, it is better, and it is simpler. But it is hard to export it to an APK, since certain devices don't support Newtonsoft.Json.dll. This DLL file would only work with Windows. In my case, sometimes I use the APIs with Meta Quest (Android). 

PS: Thanks for the contribution. 

PS2: You are right, it is more complicated, and it indeed creates formatting errors. We could add it as a unity package specifically for Windows. But you should update the package name for Windows, and delete your token.

stxrtt committed:
Add JsonUtility to serialise JSON

Class RequestWrapper is created. Unity's built-in JSON wrapper is used to serialise the entire request in one go

stxrtt commented:
Oops - thanks for pointing that out! The key was removed.

I've changed the script to first create a RequestWrapper class that will contain everything that will be serialised for the request. Then Unity's built-in JsonUtility package is used to serialise the entire request at once into JSON before sending. The Newtonsoft package was removed, so it should work on more platforms than just Windows.

What do you think?

PS: Thanks for your tutorial and project template, it's super helpful!

UnityGameStudio commented:
Indeed, it is better. And I am glad, you liked the tutorial! Thank you. I was wondering ... If I change the stream to true inside the Request Wrapper, would it work? I haven't test any streaming replies yet. It could be a good idea too, since DeepSeek takes a lot to reply. Maybe by making it streaming, it could seem that it provides "quicker" answers to the users.
